02.calendar/calendar2.rb

```diff
@@ -0,0 +1,51 @@
+#! /usr/bin/env ruby
+
+require "optparse"
+require "Date"
+
+opt = OptionParser.new
+
+option = {}
+opt.on("-m", "--month <MONTH>") {|v| option[:m] = v}
+opt.on("-y", "--year <YEAR>") {|v| option[:y] = v}
+opt.parse(ARGV)
+
+if option.size == 0
+  year = Date.today.year
+  mon = Date.today.month
+elsif option.size == 1
+  year = Date.today.year
+  mon = "#{option[:m]}".to_i
+else
+  year = "#{option[:y]}".to_i
+  mon = "#{option[:m]}".to_i
+end
+
+puts "#{mon}月 #{year}".center(20)
+
+week = ["日", "月", "火", "水", "木", "金", "土"]
+puts week.join(" ")
+
+Firstday = Date.new(year, mon, 1).day
+Lastday = Date.new(year, mon, -1).day
```

Comment:
Rubyではクラス名や定数でなければ、変数名の最初の文字を大文字にしないという暗黙のルールがあるので

```suggestion
firstday = Date.new(year, mon, 1).day
lastday = Date.new(year, mon, -1).day
```

何か意図がなければ、こんな感じにしましょう。